apps/mobile/src/features/ImportReadOnly/AddSignersForm.container.tsx

```diff
@@ -0,0 +1,90 @@
+import { View } from 'tamagui'
+import { router, useLocalSearchParams } from 'expo-router'
+import React, { useMemo } from 'react'
+import { SafeButton } from '@/src/components/SafeButton'
+import { StyleSheet } from 'react-native'
+import { SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context'
+import { makeSafeId } from '@/src/utils/formatters'
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks'
+import { selectAllChainsIds } from '@/src/store/chains'
+import { useSafesGetOverviewForManyQuery } from '@safe-global/store/gateway/safes'
+import { addSafe } from '@/src/store/safesSlice'
+import { setActiveSafe } from '@/src/store/activeSafeSlice'
+import { Address } from '@/src/types/address'
+import { groupSigners } from '@/src/features/Signers/hooks/useSignersGroupService'
+import { SignersList } from '@/src/features/Signers/components/SignersList'
+import { selectSigners } from '@/src/store/signersSlice'
+import { SignerSection } from '@/src/features/Signers/components/SignersList/SignersList'
+import { extractChainsFromSafes, extractSignersFromSafes } from '@/src/features/ImportReadOnly/helpers/safes'
+
+export const AddSignersFormContainer = () => {
+  const params = useLocalSearchParams<{ safeAddress: string }>()
+  const dispatch = useAppDispatch()
+  const chainIds = useAppSelector(selectAllChainsIds)
+  const appSigners = useAppSelector(selectSigners)
+  const { currentData, isFetching } = useSafesGetOverviewForManyQuery({
+    safes: chainIds.map((chainId: string) => makeSafeId(chainId, params.safeAddress)),
+    currency: 'usd',
+    trusted: true,
+    excludeSpam: true,
+  })
+
+  const safeAvailableOnChains = extractChainsFromSafes(currentData || [])
+  const signers = extractSignersFromSafes(currentData || [])
+  const signersGroupedBySection = useMemo(() => groupSigners(Object.values(signers), appSigners), [signers, appSigners])
+
+  const insets = useSafeAreaInsets()
+
+  const signersSections = Object.keys(signersGroupedBySection)
+    .map((group) => {
+      return signersGroupedBySection[group].data.length ? signersGroupedBySection[group] : null
+    })
+    .filter(Boolean) as SignerSection[]
+
+  return (
+    <SafeAreaView style={[styles.container, { paddingBottom: insets.bottom + insets.top }]}>
+      <SignersList
+        navbarTitle={'Import your signers to unlock account'}
+        isFetching={isFetching}
+        hasLocalSingers={!!signersGroupedBySection.imported?.data.length}
+        signersGroup={signersSections}
+      />
+
+      <View style={{ paddingHorizontal: 16 }}>
+        <SafeButton
+          onPress={() => {
+            if (!currentData) {
+              return
+            }
+            dispatch(addSafe({ SafeInfo: currentData[0], chains: safeAvailableOnChains }))
+            dispatch(
+              setActiveSafe({
+                address: currentData[0].address.value as Address,
+                chainId: currentData[0].chainId,
+              }),
+            )
```

Comment:
shall we do this second dispatch inside the `addSafe` action? otherwise we will need to call these two dispatch every time we want to add a new safe.

Reply:
I think it is ok. We don't have many places where we add a safe. This gives greater flexibility.